### ISSUE TITLE
Provide info messages for mines that don't respond when the daily update is performed. 

### ISSUE THREAD

**Issue by yochannah** [intermine/intermine-dataset-registry#1]
I'm not sure exactly why, but it looks like all the https mines aren't showing up on the app for some reason - examples include yeastmine, thalemine, legumemine, chickpeamine. @arunans23 - any idea what might be up?

**Comment by arunans23** [comment]
For [legumine](https://mines.legumeinfo.org/legumemine/service/query/results?query=%3Cquery%20name%20%3D%20%22%22%20model%3D%22genomic%22%20view%3D%22DataSet.name%20DataSet.url%20DataSet.description%20DataSet.dataSource.description%20DataSet.dataSource.name%20DataSet.dataSource.url%22%20longDescription%3D%22%22%20sortOrder%3D%22DataSet.name%20asc%22%3E%0A%3C%2Fquery%3E) and [chickpeamine](https://mines.legumeinfo.org/chickpeamine/service/query/results?query=%3Cquery%20name%20%3D%20%22%22%20model%3D%22genomic%22%20view%3D%22DataSet.name%20DataSet.url%20DataSet.description%20DataSet.dataSource.description%20DataSet.dataSource.name%20DataSet.dataSource.url%22%20longDescription%3D%22%22%20sortOrder%3D%22DataSet.name%20asc%22%3E%0A%3C%2Fquery%3E), when I run the query they aren't showing any datasets. For the case of [yeastmine](https://yeastmine.yeastgenome.org/yeastmine/service/query/results?query=%3Cquery%20name%20%3D%20%22%22%20model%3D%22genomic%22%20view%3D%22DataSet.name%20DataSet.url%20DataSet.description%20DataSet.dataSource.description%20DataSet.dataSource.name%20DataSet.dataSource.url%22%20longDescription%3D%22%22%20sortOrder%3D%22DataSet.name%20asc%22%3E%0A%3C%2Fquery%3E) and [thalemine](https://apps.araport.org/thalemine/service/query/results?query=%3Cquery%20name%20%3D%20%22%22%20model%3D%22genomic%22%20view%3D%22DataSet.name%20DataSet.url%20DataSet.description%20DataSet.dataSource.description%20DataSet.dataSource.name%20DataSet.dataSource.url%22%20longDescription%3D%22%22%20sortOrder%3D%22DataSet.name%20asc%22%3E%0A%3C%2Fquery%3E), I find that sometimes the request times out. Since the application is doing a daily job to fetch the datasets, if the request fails at that time, it stays same for the whole day until the next successful retrieve.  At this time, I see that yeastmine is listed in the dataset registry. 

I would like your opinion on what to do next.  Also I found that it is difficult to check the datasets belonging to a particular mine since the search returns results based only on name and description. I'll try to add a feature that addresses this problem.

**Comment by yochannah** [comment]
Hmmm, that makes sense. 

I wonder if a way to mitigate it would be to have a little infobox somewhere to tell people there are failed mines that are on the registry but are uncontactable. An example of how this works in a different cross-intermine app is here - note the "error loading results" on the bottom left:

![image](https://user-images.githubusercontent.com/9271438/38821341-27228cac-4198-11e8-9a5c-f9202e33ecc7.png)

What do you think?

**Comment by arunans23** [comment]
I agree. +1 for the info box suggestion. I'll add the feature in the next few days, and I'll let you know.

**Comment by yochannah** [comment]
Hey @arunans23, did the info box ever get implemented? Not meaning to rush you at all, but I was thinking that if not, I might add the hacktoberfest label to this issue and see if anyone else would like to pick it up! :)

**Comment by arunans23** [comment]
Arrrrgh!!! I completely forgot about this. @yochannah I didn't implement this yet. Please add the hacktoberfest label and see anyone is interested. Else I will do it myself and issue a pull request at the end of this month. Sorry about it.

**Comment by majumderS** [comment]
I am trying to do npm start to start the application. My mongoddb server is running at 27017. However I get this error on npm start:
**MongoDB connection error: { MongoError: failed to connect to server [undefined:27017] on first connect [MongoError: getaddrinfo ENOTFOUND undefined undefined:27017]**

Any help?

**Comment by arunans23** [comment]
@majumderS Inside the [models](https://github.com/intermine/intermine-dataset-registry/tree/master/models) folder, there are two files which are called dataset.js and lastUpdate.js. In the beginning of the file, we create the mongodb connection. Please update the Url according to your port number. (For eg: mongodb://127.0.0.1:27017/dataset && mongodb://127.0.0.1:27017/lastUpdate). Make sure you have created these collections in mongodb.

To get the data, you can then run [this](http://localhost:3000/service/update). It will fetch all the current data.

**Comment by majumderS** [comment]
Thanks @arunans23 but now that the datasets are made and connection getting through I am getting another erro MongoDB connection error: { Error: Trying to open unclosed connection.}

**Comment by majumderS** [comment]
Thanks @arunans23 that helped! :

**Comment by majumderS** [comment]
@arunans23  the frontend is only running, although I have updated the datasets using localhost:3000/service/update, I see 0 datasets in the frontend card.

**Comment by arunans23** [comment]
@majumderS make sure you have created the collections "dataset" and "lastUpdate" in mongodb.

**Comment by majumderS** [comment]
I have created collections in monodb. 
![error](https://user-images.githubusercontent.com/8430165/47152895-97311280-d2fb-11e8-87e0-a09b0be6ce26.PNG)

One query does this batchjob takes time to update the databases? I have tried to console.log at a number of places the response in service.js update route section, but did not get any result printed.

**Comment by arunans23** [comment]
@majumderS Yes, it does take some time since it has to visit all the mines to get the data. But if everything is configured correctly, all the data will be fetched and stored in around 1 minute.

**Comment by majumderS** [comment]
It does not work. My default mongodb dbpath is C:\data\db and on doing show collections, it shows lastUpdate and database.
Am I missing out something here? None of the apis get response from the server. For example
http://localhost:3000/service/datasets  =>  GET /service/datasets - - ms - -
http://localhost:3000/service/last-update => GET /service/last-update - - ms - -
This is what I keep getting. 
I am unable to get anything filled up in frontend also. 
![capture](https://user-images.githubusercontent.com/8430165/47600109-d0dbda80-d9d9-11e8-9d12-567202bc3ed7.PNG)

Novice in node and mongodb, so please help. 

Thanks.

**Comment by arunans23** [comment]
@majumderS  are you getting any error in the console when you run "localhost:3000/service/update". If so please paste it here.

I created a new instance and tested everything from the beginning. Everything is working fine for me. Please make sure you have done the following.

1. Changed the URL in the dataset.js and lastUpdate.js files. Eg:
`mongoose.connect("mongodb://127.0.0.1:27017/lastUpdate");`
(Mind the quotes inside the connect())

2. Create the mongodb collections. (if everything works properly you dont even need to create the collections manually. In the update process, everything will get created automatically)

3. Make sure you ran the update service. "http://localhost:3000/service/update"

**Comment by majumderS** [comment]
Hi @arunans23 Sorry for the delay in response. It works fine now. Actually I was getting a MongoDB connection error: { Error: Trying to open unclosed connection because the code has mongoose.connect("mongodb://127.0.0.1:27017/dataset");
I tried to handle it by replacing it with mongoose.createConnection("mongodb://127.0.0.1:27017/dataset");
That is the reason, that nothing was getting loaded. But now although the error shows up, the app also runs fine.
![interminedatasets img](https://user-images.githubusercontent.com/8430165/47811269-5f01e900-dd6b-11e8-97e9-354dd8895bfc.PNG)


Yay! 

Thanks for all the help. 

Btw any idea why the above error comes? Also I would like to work on the issue,  As I can see now, if any mine is not fetchable, no msg is shown. It is simply blank. I went through the previous chats and I think I can try implementing a infobox as you guys are suggesting.

**Comment by majumderS** [comment]
Hi @arunans23 so I was working on the infobox issue, and first of all I tried to get a hang of the code and next I implemented a running feature of the suggested changes i.e. some info-section if dataset is not being able to get connected.
I will work on the design part as well, but just have a look below and let me know if this is what you are looking for.

![no_data](https://user-images.githubusercontent.com/8430165/47813988-8f00ba80-dd72-11e8-8416-78f3c1b40f9f.PNG)

Let me know.

**Comment by yochannah** [comment]
@majumderS I think this is a good start! Could we rephrase it a little to make things clearer, though? Perhaps something like 

"No data available for ThaleMine - [try visiting ThaleMine directly for more info](https://apps.araport.org/thalemine)"
